File: 1lab/py/d.py
str = input()
list = []
map = {}
for i in str:
    if i in list:
        temp = str.index(i)
        if len(list) > 0 and str[temp] == list[len(list) - 1]:
            list.pop()
    else:
        list.append(i)
for i in str:
    if i not in map:
        map[i] = 1
    else: 
        map[i] += 1
for i in map:
    if map[i] % 2 != 0:
        print("NO")
        exit()
if len(list)==0:
    print("YES")
else:
    print("NO")

File: 1lab/d.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    string s; cin>>s;
    stack <char> st;
    int n = s.size();
    bool flag = false;
    for (int i = 1;i < n ; i++){
        st.push(s[i - 1]);
        if (flag == true){
            st.pop();
        }
        if (s[i] == st.top()){
            st.pop();
            flag = true;
        }
    }

    if (st.empty() == true){
        cout<<"Yes";
    }
    else cout<<"No";
    return 0;
}